### PR TITLE
When interpolating a string with a List variable, use the textual representation of the List

### PR DESCRIPTION
**Pull request by yegappan**

### CODE REVIEW

File: src/eval.c [hunk 1]
```diff
@@ -597,7 +597,9 @@ typval2string(typval_T *tv, int convert)
 	ga_append(&ga, NUL);
 	retval = (char_u *)ga.ga_data;
     }
-    else if (convert && tv->v_type == VAR_DICT)
+    else if (tv->v_type == VAR_LIST)
+	retval = list2string(tv, get_copyID(), FALSE);
+    else if (tv->v_type == VAR_DICT)
 	retval = dict2string(tv, get_copyID(), FALSE);
```
Comment (zeertzjq) [inline src/eval.c]: Maybe this should use `tv2string()` instead? That handles NULL list and NULL dict.

**Comment by zeertzjq** [comment]
The argument name `convert` also seems no longer meaningful. Maybe it should be renamed `join_list` instead.

**Comment by zeertzjq** [comment]
The argument name `convert` in `typval2string()` and `eval_to_string()` also seems no longer meaningful. Maybe it should be renamed to `join` instead.

File: src/eval.c (hunk 1 above)
Comment (yegappan) [reply src/eval.c]: Thanks.  Modified the code to use tv2string() for Dict and List.  Add additional tests for NULL List/Dict.

**Comment by yegappan** [comment]
> The argument name `convert` in `typval2string()` and `eval_to_string()` also seems no longer meaningful. Maybe it should be renamed to `join` instead.

Renamed the argument from "convert" to "join_list"..

**Comment by chrisbra** [comment]
thanks!
